### PR TITLE
Extend `tokenizer.pad` with `offset_mapping`

### PR DESCRIPTION
**Pull request by vadimirtlach**
# What does this PR do?

Extend `tokenizer.pad`, accurately `tokenizer._pad`, method with functionality for padding `offset_mapping` with `(0, 0)`.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [Padding offsets mapping via tokenizer.pad #18681
](https://github.com/huggingface/transformers/issues/18681)? Please add a link
      to it if that's the case.
- [] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by ydshieh** [comment]
Hi @vad13irt. Could you explain a bit why this change is needed, and what would be its usage? With a short code snippet might be very helpful. Thank you!

**Comment by LysandreJik** [comment]
cc @SaulLu  :)

File: src/transformers/tokenization_utils_base.py [hunk 1]
```diff
@@ -3299,7 +3299,8 @@ def _pad(
             max_length = ((max_length // pad_to_multiple_of) + 1) * pad_to_multiple_of
 
         needs_to_be_padded = padding_strategy != PaddingStrategy.DO_NOT_PAD and len(required_input) != max_length
-
+        special_token_offset = (0, 0)
```
Comment (SaulLu) [inline src/transformers/tokenization_utils_base.py:3302]: A small nit, but I would give it another name to designate that it is the offset mapping of the pad token. 

How about `pad_token_offset`?

**Review by SaulLu — COMMENTED** [review]
Thank you very much for your contribution @vad13irt. :hugs: 

As mentioned in issue  #1868, a very important element to be able to merge this type of PR - which will concern the tokenizers of all the models having a fast version - is to add a test that would have failed before this PR and which passes with this PR. A good place to add this test would be in the [tests/test_tokenization_common.py](https://github.com/huggingface/transformers/blob/main/tests/test_tokenization_common.py) file.

A small note regarding @ydshieh 's question: indeed to pad and truncate with a fast tokenizer it is better to do it during the encoding of the example and not afterwards (because it will use the rust backend). However, it's true that arguments specific to fast tokenizers - like `offset_mapping` - could also be handled by the pad method which allows to pad an already encoded text.

File: src/transformers/tokenization_utils_base.py (hunk 1 above)
Comment (vad13irt) [reply src/transformers/tokenization_utils_base.py:3302]: Yeah, good variant! Do I need something to do yet, e.g tests?

Comment (SaulLu) [reply src/transformers/tokenization_utils_base.py:3302]: > Do I need something to do yet, e.g tests?

Yes indeed these changes will have to be included in this PR :relaxed:

Comment (vad13irt) [reply src/transformers/tokenization_utils_base.py:3302]: Ok

**Pull request closed without merging**